git_test.go

```diff
@@ -207,153 +207,217 @@ func (r *FakeGitRepo) read() *GitScm {
 			os.Setenv("CI_MERGE_REQUEST_TARGET_BRANCH_NAME", "main")
 			defer restoreGitlabFn()
-			defer restoreGithubFn()
 
-			sha, baseRef, provider, changeRequest := checkGitProvider()
-			assert.Equal(t, "0123456", sha)
-			assert.Equal(t, "main", baseRef)
-			assert.Equal(t, "Gitlab", provider)
-			assert.True(t, changeRequest)
+			gitCtx := checkGiContext()
+			assert.Equal(t, "0123456", gitCtx.Commit)
+			assert.Equal(t, "branch", gitCtx.Branch)
+			assert.Equal(t, "main", gitCtx.GetTargetBranch())
+			assert.Equal(t, "Gitlab", gitCtx.Provider)
+			assert.True(t, gitCtx.ChangeRequest)
 		})
 	})
 
 	t.Run("Local machine", func(t *testing.T) {
-		if originalSha != "" {
-			t.Skip("Tests skipped when running on Github")
-		}
-
-		cleanGithubFn()
-
 		t.Run("Running with TARGET_BRANCH", func(t *testing.T) {
+			os.Setenv("BRANCH", "foo")
 			os.Setenv("TARGET_BRANCH", "main")
-			defer restoreGithubFn()
 			defer os.Unsetenv("TARGET_BRANCH")
-
-			sha, baseRef, provider, changeRequest := checkGitProvider()
-			assert.Equal(t, "", sha)
-			assert.Equal(t, "main", baseRef)
-			assert.Equal(t, "", provider)
-			assert.False(t, changeRequest)
+			defer os.Unsetenv("BRANCH")
+			defer restoreSCMContextsFn()
+
+			gitCtx := checkGiContext()
+			assert.Equal(t, "", gitCtx.Commit)
+			assert.Equal(t, "foo", gitCtx.Branch)
+			assert.Equal(t, "main", gitCtx.GetTargetBranch())
+			assert.Equal(t, "", gitCtx.Provider)
+			assert.True(t, gitCtx.ChangeRequest)
 		})
 
 		t.Run("Running without TARGET_BRANCH", func(t *testing.T) {
-			defer restoreGithubFn()
-
-			sha, baseRef, provider, changeRequest := checkGitProvider()
-			assert.Equal(t, "", sha)
-			assert.Equal(t, "", baseRef)
-			assert.Equal(t, "", provider)
-			assert.False(t, changeRequest)
+			os.Setenv("BRANCH", "foo")
+			defer os.Unsetenv("BRANCH")
+			restoreSCMContextsFn()
+
+			gitCtx := checkGiContext()
+			assert.Equal(t, "", gitCtx.Commit)
+			assert.Equal(t, "foo", gitCtx.Branch)
+			assert.Equal(t, "foo", gitCtx.GetTargetBranch())
+			assert.Equal(t, "", gitCtx.Provider)
+			assert.False(t, gitCtx.ChangeRequest)
 		})
 	})
+
+	t.Run("Empty SCM context", func(t *testing.T) {
+		unsetGithub()
+		defer restoreSCMContextsFn()
+
+		gitCtx := checkGiContext()
+		assert.Nil(t, gitCtx)
+	})
 }
 
 func TestGit_ContributeAttributesCloneOptions(t *testing.T) {
-	os.Setenv("TARGET_BRANCH", "master") // master branch is the base branch for the fake repository (octocat/hello-world)
-	defer os.Unsetenv("TARGET_BRANCH")
+	os.Setenv("BRANCH", "master") // master branch is the base branch for the fake repository (octocat/hello-world)
+	defer func() {
+		os.Unsetenv("BRANCH")
+	}()
 
-	scm := NewFakeGitRepo(t, WithCloneOptions(CloneOptionsRequest{Depth: 1})).withBranch("this-is-a-test-branch").addingFile("TEST-sample2.xml").removingFile("README").withCommit("This is a test commit").read()
+	scm := NewFakeGitRepo(t, WithCloneOptions(CloneOptionsRequest{Depth: 1})).read()
```

Comment:
This is a simplification, as there is no need to add commits to calculate the clone options